components/Authentication/signup.jsx

```diff
@@ -115,7 +118,151 @@ const Signup = ({ navigation }) => {
+                                {time: 10, weight: 60},
+                              ]},
+                            {
+                              exerciseID: "An8hwIGvJrMMphdxubUs",
+                              sets: [
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                              ]
+                            },
+                            {
+                              exerciseID: "X9HKNuWTf5zTHqpobfxS",
+                              sets: [
+                                {reps: 12, weight: 100},
+                                {reps: 12, weight: 100},
+                                {reps: 12, weight: 100},
+                              ]
+                            }
+                          ]
+            }
+
+            const legsWorkout = {
+              name: "Legs",
+              muscleGroups: "Quads Glutes",
+              lastPerformed: currentTime,
+              exercises: [
+                            {
+                              exerciseID: "31ROy02NIqplIBvXoaeB", 
+                              sets: [
+                                {time: 10, weight: 135},
+                                {time: 10, weight: 135},
+                                {time: 10, weight: 135},
+                              ]},
+                            {
+                              exerciseID: "2TvJvGO8CuxXzxk1D2Si",
+                              sets: [
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                              ]
+                            },
+                            {
+                              exerciseID: "XzkLscitllVWr1sRrMAk",
+                              sets: [
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                                {reps: 10, weight: 100},
+                              ]
+                            }
+                          ]
+            }
+
+            //set initial user 
+            const userRef = db.collection('users').doc(user.user.uid);
+            
+            const userDataRef = userRef.set(userData);
+            const userWorkout1 = userRef.collection('workouts').doc();
+            const userWorkout2 = userRef.collection('workouts').doc();
+            const userWorkout3 = userRef.collection('workouts').doc();
+
+            Promise.all(
+              userDataRef,
+              userWorkout1.set(pushWorkout), 
+              userWorkout2.set(pullWorkout),
+              userWorkout3.set(legsWorkout)
+            ).then(() => {
+              console.log("workouts and user set successfully.");
+            }).catch((error) => {
+              console.log(error);
+            });
```

Comment:
Can we make this an async function and use await on these promises and the other promises. The reason why I say this is right now we're navigating to home **before** Firebase is done setting up a user's information. 

Not sure if this is the issue but I hope it is cause its an easy fix. But when I create a new user and it logs me in right away, my app is empty and my firebase responses are empty as well. I'm guessing its because the info isn't set in firebase yet so we need to wait for it to finish setting up and then navigate to home.